controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1122,6 +1124,177 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+                    neighbor_al_mac,
+                    RxLinkMetricData
+                        ->al_mac_of_the_neighbor_whose_link_metric_is_reported_in_this_tlv())) {
+                LOG(ERROR)
+                    << "TLV_RECEIVER_LINK_METRIC reported al_mac ="
+                    << network_utils::mac_to_string(neighbor_al_mac) << std::endl
+                    << " and TLV_TRANSMITTER_LINK_METRIC reported al_mac ="
+                    << network_utils::mac_to_string(
+                           RxLinkMetricData
+                               ->al_mac_of_the_neighbor_whose_link_metric_is_reported_in_this_tlv())
+                    << std::endl
+                    << " not the same";
+                return false;
+            }
+
+            LOG(DEBUG) << "recieved  tlvReceiverLinkMetric from al_mac="
+                       << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                       << "reported  al_mac =" << network_utils::mac_to_string(neighbor_al_mac)
+                       << std::endl;
+
+            //fill rx data from TLV
+            if (!new_link_metric_data.add_receiver_link_metric(RxLinkMetricData)) {
+                LOG(ERROR) << "adding RxLinkMetricData has failed";
+                return false;
+            }
+        }
+    }
+
+    //save agent link metric data to database or modify existing.
+    if (m_link_metric_data.find(reporting_agent_al_mac) != m_link_metric_data.end()) {
+        if (m_link_metric_data[reporting_agent_al_mac].find(neighbor_al_mac) !=
+            m_link_metric_data[reporting_agent_al_mac].end()) {
+            m_link_metric_data[reporting_agent_al_mac][reporting_agent_al_mac] =
+                new_link_metric_data;
+            LOG(DEBUG) << "metric data update for existing al_mac = "
+                       << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                       << " updating old report from neighbor al_mac ="
+                       << network_utils::mac_to_string(neighbor_al_mac) << std::endl;
+        } else {
+            m_link_metric_data[reporting_agent_al_mac].insert(
+                std::make_pair(neighbor_al_mac, new_link_metric_data));
+            LOG(DEBUG) << "metric data update for existing al_mac = "
+                       << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                       << " adding new report for neighbor al_mac ="
+                       << network_utils::mac_to_string(neighbor_al_mac) << std::endl;
+        }
+    } else {
+        m_link_metric_data[reporting_agent_al_mac].insert(
+            std::make_pair(neighbor_al_mac, new_link_metric_data));
+        LOG(DEBUG) << "Add metric data from new al_mac = "
+                   << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                   << " adding new report for neighbor al_mac ="
+                   << network_utils::mac_to_string(neighbor_al_mac) << std::endl;
+    }
```

Comment:
Correct me if I am wrong, but AFAIU, what you are trying to do here is to update the container from the database with new data, and if it doesn't have a specific mac, add it. In that case, all this code here is useless since `[]` operator does already insert a new element if it doesn't exist. This means that you could replace all of this code with one line:
`link_metric_data[reporting_agent_al_mac][reporting_agent_al_mac] = new_link_metric_data;`

Reply:
1. I would like to know what really happens in case of a failure.
2. in the future i'm not sure that we would like to update the results in all cases, maybe we would like to analyze it first.

in any case i think that it doesn't hurt and the code is more readable.

Reply:
But there is no failure. It is always, insertion of a new element or updates one, and I don't see how it is interesting which one has occurred.
Leaving code for uncertain future purpose is a bad idea. If it doesn't contribute anything it better be removed.

Reply:
You need to understand that there is more than writing a pretty code.
It has to be functional.
We are in the middle of certification testing with multiple golden units.
Our job is to identify vulnerable spots and add prints that will be helpful for debugging issues like M3 msg with Qualcom agent.
Also,  this is temporary code that will be replaced by the actual functionality in further stages.